### DISCUSSION TITLE
Appropriate project for GitHub?

### DISCUSSION THREAD

**Discussion by James-Brown** [community/community#39784] [New to GitHub]
I want to start a repository to support my project, but I wan't to understand if this project will work on this system. So before I invest a lot of time and energy moving to GitHub, I would like to ask a few questions:

The project is the development of a Neuron Simulator. I have it up and running now. Is this a good time to move it, or am I too late?

It is coded in Delphi (Pascal) and contains many proprietary components. Of these, I have source code for some not. Of course, I would not post the components, but how would I inform a user about the need, type, and origin of these?

The application is linked to a website [ www.SETI.Net ] that contains most of the help files. The application is also linked to YouTube videos that describe its operation. Will this work on GitHub?

Can I post the compiled application as an .EXE for download? What about the installer files (I use Inno Setup)? Need they be included?

The simulator is experimental, so I need feedback from experts in the field (neurologists, computational neuro simulator experts, etc.). Is this supported somehow in the system?

Thanks, Jim

**Comment by swastkk** [comment]
Hello @James-Brown , to my knowledge I think you should have your Neuron Simulator project on GitHub. Github will have all your 
 Files , will provide you the version control plus you can have all your files here whether it is .exe file for the download purpose and the hosted site configuration files which will be very much handy because you will simply share the repo links with others where they can use your Application and access the Download files too. And for  how to use the Application, you can have Readme file too in the Repo for the ease of usage to the community.
This is my opinion regarding your doubt , others will also provide you the worthy inputs .
Thanks, Swastik

**Comment by airtower-luna** [comment]
> It is coded in Delphi (Pascal) and contains many proprietary components. Of these, I have source code for some not. Of course, I would not post the components, but how would I inform a user about the need, type, and origin of these?

Include a README that describes these things, and possibly links to further documentation (for example the other resources you mentioned).

> Can I post the compiled application as an .EXE for download? What about the installer files (I use Inno Setup)? Need they be included?

Do *not* include binaries in the repository itself. You can upload them as [release assets](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases) to make them available.

> The simulator is experimental, so I need feedback from experts in the field (neurologists, computational neuro simulator experts, etc.). Is this supported somehow in the system?

There's infrastructure for providing comments, feedback, and suggested changes, yes. People can write issues, PRs, and (if enabled) discussions (like this one) in your repository. There isn't any built-in expert panel, though. I'm not familiar with those fields, so I'm not sure how to best catch the interest of experts who might have time and motivation to take a look.
